datatypes/src/primitives/coordinate.rs

```diff
@@ -2,16 +2,19 @@ use crate::util::arrow::ArrowTyped;
 use arrow::array::{BooleanArray, Float64Builder};
 use arrow::datatypes::DataType;
 use arrow::error::ArrowError;
+use ocl::OclPrm;
 use serde::{Deserialize, Serialize};
 use std::{fmt, slice};
 
-#[derive(Clone, Copy, Debug, Deserialize, PartialEq, PartialOrd, Serialize)]
+#[derive(Clone, Copy, Debug, Deserialize, PartialEq, PartialOrd, Serialize, Default)]
 #[repr(C)]
 pub struct Coordinate2D {
     pub x: f64,
     pub y: f64,
 }
 
+unsafe impl OclPrm for Coordinate2D {}
```

Comment:
What is this? Maybe you could write a comment on that.

Reply:
Is this an OpenCL parameter? why is it needed?

Reply:
it's a marker trait that signals data of this type can be transferred to the device. Only works for repr-c types

Reply:
91ef24f5b869f75563586b32b932afa0a27fce0f